### PR TITLE
Dropdown - don't stop event propagation

### LINKED ISSUE #30267: Dropdown data-toggle="dropdown" element click events not bubble
_addEventListeners function has two functions to prevent default link actions.
      event.preventDefault()
      event.stopPropagation()

If link element is not used for toggle, disabling events are not nesessary at all?

Reason is i need attach events to all toggle elements to get click events wanted from toggle elements. 
Better way is to attach single event listener to parent and use event.target, this is not working now.

Is it possible to add check if toggle element is link, then add those events something like:
`
if(this._element.nodeName == 'a'){
      event.preventDefault();
      event.stopPropagation();
}
`
or using data attribute to disable/enable bubling.

### PR DESCRIPTION
**Pull request by sijusamson**
Fixes #30267

Preview: https://deploy-preview-32943--twbs-bootstrap.netlify.app/

### CODE REVIEW

**Comment by XhmikosR** [comment]
We also have #32625, /CC @rohit2sharma95

**Comment by rohit2sharma95** [comment]
It is about another issue that is when the dropdown toggle button is clicked, the (click) event is not bubbling up because of the `stopPropagation`.

**Comment by mohe2015** [comment]
Does this also fix https://github.com/twbs/bootstrap/issues/30882 ?

**Comment by XhmikosR** [comment]
@sijusamson can you add a test please?

**Comment by XhmikosR** [comment]
@sijusamson could this possibly fix #33044? If so, can you add another test case covering that too please?

**Comment by sijusamson** [comment]
> @sijusamson could this possibly fix #33044? If so, can you add another test case covering that too please?

Not sure. I will have a look.
And I am so sorry for reverting too late :-(

File: js/tests/unit/dropdown.spec.js [hunk 1]
```diff
@@ -1719,9 +1719,7 @@ describe('Dropdown', () => {
       '  <div class="dropdown-menu">',
       '    <a class="dropdown-item disabled" href="#sub1">Submenu 1</a>',
       '    <button class="dropdown-item" type="button" disabled>Disabled button</button>',
-      '    <a id="item1" class="dropdown-item" href="#">Another link</a>',
-      '  </div>',
-      '</div>'
+      '    <a id="item1" class="dropdown-item" href="#">Another link</a>'
```
Comment (XhmikosR) [inline js/tests/unit/dropdown.spec.js:1722]: This change seems unrelated?

File: js/tests/unit/dropdown.spec.js [hunk 2]
```diff
@@ -1756,7 +1754,34 @@ describe('Dropdown', () => {
         handleArrowUp()
       }
     })
-
```
Comment (XhmikosR) [inline js/tests/unit/dropdown.spec.js:1759]: This one also

File: js/tests/unit/dropdown.spec.js [hunk 3]
```diff
@@ -1756,7 +1754,34 @@ describe('Dropdown', () => {
...
+      expect(triggerDropdown.classList.contains('show')).toEqual(true)
+      expect(dropdown.show).toHaveBeenCalled()
+      expect(event.target).toEqual(triggerDropdown)
+      done()
+    })
+
+    triggerDropdown.click()
+  })
 })
+
```
Comment (XhmikosR) [inline js/tests/unit/dropdown.spec.js:1787]: And this one

**Comment by XhmikosR** [comment]
Hmm, this seems to break at least our docs version dropdown

image

**Comment by sijusamson** [comment]
> Hmm, this seems to break at least our docs version dropdown
> 
> image

Great catch. :100:

**Comment by XhmikosR** [comment]
@sijusamson can you please add an extra test so that the previous regression is caught? Because I only caught the issue manually.

**Review by sijusamson — COMMENTED** [review]
@XhmikosR I have added a test case for the same. The issue was caused because a child element was present within the dropdown. 
I have handled the case as well as added a test case too.

**Review by rohit2sharma95 — APPROVED** [review]

**Comment by rohit2sharma95** [comment]
It does not fix #33044 though

**Comment by XhmikosR** [comment]
@sijusamson can you have a look in case you can also fix #33044 too here?

**Comment by sijusamson** [comment]
> @sijusamson can you have a look in case you can also fix #33044 too here?

Sure !

**Comment by XhmikosR** [comment]
Actually wait, it seems #32942 might be a cleaner solution :)

**Comment by sijusamson** [comment]
> Actually wait, it seems #32942 might be a cleaner solution :)

Okay.

**Comment by sijusamson** [comment]
Closing this PR as #32942 appears to be a better solution.

**Comment by arcanisgk** [comment]
by default there should be no blocking of events in the use of dropdown box; but if you want to add it programmatically it should be possible to insert a block of events.

What would happen if you want to track where the user navigates through the dropdown boxes and the events are blocked ...
